lib/sql.ts

```diff
@@ -259,12 +263,154 @@ export class SQLDatabaseManager extends QueryProcessManager<DatabaseQuery, any>
 	}
 }
 
+export const tables = new Map<string, DatabaseTable<any>>();
+
+export class DatabaseTable<T> {
```

Comment:
Shouldn't there be a way to override this - what if some rows in a table contain integers while others contain strings?

Reply:
Why would that be happening?
That seems like a failure on the part of the user. 
(Also, I'd say just typecast the query results manually where necessary).

Reply:
Sorry, I meant columns, not row.

Reply:
This API is modeled on my own library in:

https://github.com/Zarel/telepic/tree/master/server

In mine, `selectAll<T>` is generic:

https://github.com/Zarel/telepic/blob/2c55b7b09d95491e51853d1822ba990193c94c81/server/users.ts#L56

Reply:
> Shouldn't there be a way to override this - what if some column in a table contain integers while others contain strings?

Oh! I understand now. T should generally be an object, and it types the full object. IE, using your example, then we'd selectAll('*'), which returns {log_line: string, timestamp: number}[]. Even if you only select `log_line`, that still returns `{log_line: string}`.

Reply:
> In mine, `selectAll<T>` is generic:
> 
> https://github.com/Zarel/telepic/blob/2c55b7b09d95491e51853d1822ba990193c94c81/server/users.ts#L56
> 
> I'd recommend that here, too.

Can we not use a regular `query` function as a catch-all for that instead? (Or, rather, since it's SQLite - `get`, `all`, `run`, and `exec`?)

Reply:
Yes, that sounds good.

Reply:
I added type overrides to most functions (`all`, `selectAll`, `selectOne`, etc)